lib/libc/newlib/CMakeLists.txt

```diff
@@ -26,12 +26,10 @@ if(LIBC_LIBRARY_DIR)
   set(LIBC_LIBRARY_DIR_FLAG -L${LIBC_LIBRARY_DIR})
 endif()
 
-# Define _ANSI_SOURCE in order to prevent Newlib from defining POSIX primitives
-# in its headers when GNU dialect is used (-std=gnu*). Newlib features.h
-# defines _DEFAULT_SOURCE when __STRICT_ANSI__ is not defined by GCC (i.e. when
-# -std=gnu*), which leads to various POSIX definitions being provided by the
-# Newlib headers and conflicts with the POSIX definitions provided by Zephyr.
-zephyr_compile_definitions(_ANSI_SOURCE)
+# define _POSIX_C_SOURCE to pull in a few things from the
+# posix definitions that are used like strnlen, and how
+# strerror_r is defined.
+zephyr_compile_options(-D_POSIX_C_SOURCE=200809)
```

Comment:
How does this differ from the default newlib set of APIs? Picolibc offers a very similar API suite and we're not doing this for that library.

Reply:
We are doing this from picolibc, we do it here:
https://github.com/zephyrproject-rtos/picolibc/blob/main/zephyr/zephyr.cmake#L86